### DISCUSSION TITLE
Update Supabase template to use any S3-Provider for Object Storage

### DISCUSSION THREAD

**Discussion by actraiser** [coollabsio/coolify#2038] [Service Template Requests]
[The current Supabase template](https://github.com/coollabsio/coolify/blob/main/templates/compose/supabase.yaml) preconfigures MINIO as S3-Object-Storage but in regards to static files, I believe it would be beneficial to easily configure any S3 Provider. For that to work, one must right now edit the docker-compose file directly.

To make it easier to use an external S3-Provider, the following env variables have to be exposed:

STORAGE_S3_BUCKET (default: stub)
STORAGE_S3_REGION (default: us-east-1)
STORAGE_S3_ENDPOINT (default: http://supabase-minio:9000)

Also, for clarifying that the provided Minio-Service is the out-of-the-box solution to **self-host** objects, but that actually any S3-Provider would work, the env variables SERVICE_USER_MINIO and SERVICE_PASSWORD_MINIO could be renamed to STORAGE_S3_ACCESS_KEY_ID and STORAGE_S3_SECRET_ACCESS_KEY

Greets
-act

**Comment by mattmarcin** [comment]
@actraiser Would you be able to supply a little tutorial here on how to edit the Dockerfile while we wait for a change?

**Reply by actraiser** [reply]
I overwrote three values in the docker-compose file that otherwise would point to the internal Minio Container that Coolify provides. The Minio User Name and Minio Password, which are generated by Coolify when deploying the Supabase template, can be overriden directly in the Coolify UI.  They represent ACCESS_KEY_ID and SECRET_ACCESS_KEY  for your S3-Provider. You could of course also create environment variables instead of hardcoding the values into the docker-compose file. However, when upgrading the Supabase template, I suppose the docker-compose file is re-created based on the latest template and the auto-generated Minio-Values for Endpoint, Bucket and Region would show up again.

So no matter what, it would be beneficial to make the STORAGE_S3_BUCKET,  STORAGE_S3_ENDPOINT and STORAGE_S3_REGION configurable in the Coolify UI at some point,, so custom S3-Configuration survives upgrades of the Supabase service.

```
- STORAGE_S3_BUCKET=<bucket_name> // bucket name, e.g. myBucket
- STORAGE_S3_ENDPOINT=<s3_endpoint> // e.g. https://<id>.r2.cloudflarestorage.com
- STORAGE_S3_REGION=<region> // us-east-1 etc
- AWS_ACCESS_KEY_ID=${SERVICE_USER_MINIO}  // no need to touch, change in UI
- AWS_SECRET_ACCESS_KEY=${SERVICE_PASSWORD_MINIO} // no need to touch, change in UI
```
     
-act

**Reply by gabrielsebag** [reply]
For anyone struggling to integrate with Amazon S3, here’s the correct configuration:
```
STORAGE_S3_BUCKET=<bucket_name> // e.g. my-bucket
STORAGE_S3_ENDPOINT=https://s3.<region>.amazonaws.com // e.g. https://s3.us-east-1.amazonaws.com
STORAGE_S3_REGION=<region> // e.g. us-east-1
AWS_ACCESS_KEY_ID=${SERVICE_USER_MINIO}
AWS_SECRET_ACCESS_KEY=${SERVICE_PASSWORD_MINIO}
```
I initially ran into issues because I configured the endpoint like this:
```
STORAGE_S3_ENDPOINT=https://<bucket_name>.s3.<region>.amazonaws.com
```
This is incorrect — the bucket name should not be included in STORAGE_S3_ENDPOINT.

What made this especially confusing is that the setup appeared to work at first: I was able to upload files to my S3 bucket without any errors. However, problems surfaced later:

- Files could not be moved.
- Files were not deleted from S3 after being successfully deleted from Supabase.

Once I switched to the correct endpoint format (region-only, no bucket name), all of these issues were resolved.
